internal/keycloakb/accreditationsmodule.go

```diff
@@ -0,0 +1,158 @@
+	return &accredsModule{
+		keycloakClient: keycloakClient,
+		confDBModule:   confDBModule,
+		logger:         logger,
+	}
+}
+
+func (am *accredsModule) GetUserAndPrepareAccreditations(ctx context.Context, accessToken, realmName, userID, condition string) (kc.UserRepresentation, int, error) {
+	var kcUser kc.UserRepresentation
+
+	// Gets the realm
+	var realm, err = am.keycloakClient.GetRealm(accessToken, realmName)
+	if err != nil {
+		am.logger.Warn(ctx, "msg", "getKeycloakRealm: can't get realm from KC", "err", err.Error())
+		return kcUser, 0, errorhandler.CreateInternalServerError("keycloak")
+	}
+
+	// Retrieve admin configuration from configuration DB
+	var rac configuration.RealmAdminConfiguration
+	rac, err = am.confDBModule.GetAdminConfiguration(ctx, *realm.Id)
+	if err != nil {
+		am.logger.Warn(ctx, "msg", "CreateAccreditations: can't get admin configuration", "err", err.Error())
+		return kcUser, 0, errorhandler.CreateInternalServerError("keycloak")
+	}
+
+	// Evaluate accreditations to be created
+	var newAccreds []string
+	for _, modelAccred := range rac.Accreditations {
+		if modelAccred.Condition == nil || *modelAccred.Condition == condition {
+			var expiry *string
+			expiry, err = am.convertDurationToDate(ctx, *modelAccred.Validity)
+			if err != nil {
+				return kcUser, 0, err
+			}
+			var newAccreditationJSON, _ = json.Marshal(AccreditationRepresentation{
+				Type: modelAccred.Type,
+				Exp:  expiry,
+			})
+			newAccreds = append(newAccreds, string(newAccreditationJSON))
+		}
+	}
+
+	// Get the user from Keycloak
+	kcUser, err = am.keycloakClient.GetUser(accessToken, realmName, userID)
+	if err != nil {
+		am.logger.Warn(ctx, "msg", "CreateAccreditations: can't get Keycloak user", "err", err.Error(), "realm", realmName, "user", userID)
+		return kcUser, 0, err
+	}
+
+	// Update attributes in kcUser
+	// If no new accreditation, return
+	var added = 0
+	if len(newAccreds) > 0 {
+		var kcAccreds = kcUser.GetAttribute(constants.AttrbAccreditations)
+		for _, newAccred := range newAccreds {
+			if !arrayContains(kcAccreds, newAccred) {
+				kcAccreds = append(kcAccreds, newAccred)
+				added++
+			}
+		}
+		kcUser.SetAttribute(constants.AttrbAccreditations, kcAccreds)
+	}
+
+	return kcUser, added, nil
+}
+
+func (am *accredsModule) CreateAccreditations(ctx context.Context, accessToken, realmName, userID, condition string) error {
+	var kcUser, count, err = am.GetUserAndPrepareAccreditations(ctx, accessToken, realmName, userID, condition)
+	if err != nil || count == 0 {
+		return err
+	}
+
+	// Update user in Keycloak
+	err = am.keycloakClient.UpdateUser(accessToken, realmName, userID, kcUser)
+	if err != nil {
+		am.logger.Warn(ctx, "msg", "CreateAccreditations: can't update Keycloak user", "err", err.Error(), "realm", realmName, "user", userID)
+		return err
+	}
+
+	return nil
+}
+
+func arrayContains(slice []string, value string) bool {
```

Comment:
used again here

Reply:
I removed this function as validation.IsStringInSlice(...) is now merged in common-service